master/internal/db/postgres_groups.go

```diff
@@ -0,0 +1,218 @@
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		return matchSentinelError(err)
+	}
+
+	return nil
+}
+
+// UpdateGroup updates a group in the database. Returns ErrNotFound if the
+// group isn't found.
+func (db *PgDB) UpdateGroup(ctx context.Context, group model.Group) error {
+	res, err := Bun().NewUpdate().Model(&group).WherePK().Exec(ctx)
+
+	return matchSentinelError(mustHaveAffectedRows(res, err))
+}
+
+// AddUsersToGroup adds users to a group by creating GroupMembership rows.
+// Returns ErrNotFound if the group isn't found or ErrDuplicateRow if one
+// of the users is already in the group.
+func (db *PgDB) AddUsersToGroup(ctx context.Context, gid int, uids ...model.UserID) error {
+	if len(uids) < 1 {
+		return nil
+	}
+
+	groupMem := make([]model.GroupMembership, 0, len(uids))
+	for _, uid := range uids {
+		groupMem = append(groupMem, model.GroupMembership{
+			UserID:  uid,
+			GroupID: gid,
+		})
+	}
+
+	tx, err := Bun().BeginTx(ctx, nil)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		rbErr := tx.Rollback()
+		if rbErr != sql.ErrTxDone && rbErr != nil {
+			log.WithError(rbErr).
+				WithField("groupID", gid).
+				WithField("userIDs", uids).
+				Error("error rolling back transaction in AddUsersToGroup")
+		}
+	}()
+
+	res, err := tx.NewInsert().Model(&groupMem).Exec(ctx)
+	if foundErr := mustHaveAffectedRows(res, err); foundErr != nil {
+		return matchSentinelError(foundErr)
+	}
+
+	err = tx.Commit()
+	return matchSentinelError(err)
+}
+
+// RemoveUsersFromGroup removes users from a group. Removes nothing and
+// returns ErrNotFound if the group or one of the users' membership rows
+// aren't found.
+func (db *PgDB) RemoveUsersFromGroup(ctx context.Context, gid int, uids ...model.UserID) error {
+	if len(uids) < 1 {
+		return nil
+	}
+
+	tx, err := Bun().BeginTx(ctx, nil)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		rbErr := tx.Rollback()
+		if rbErr != sql.ErrTxDone && rbErr != nil {
+			log.WithError(rbErr).
+				WithField("groupID", gid).
+				WithField("userIDs", uids).
+				Error("error rolling back transaction in RemoveUsersFromGroup")
+		}
+	}()
+
+	res, err := tx.NewDelete().Table("user_group_membership").
+		Where("group_id = ?", gid).
+		Where("user_id IN (?)", bun.In(uids)).
+		Exec(ctx)
+	if foundErr := mustHaveAffectedRows(res, err); foundErr != nil {
+		return matchSentinelError(foundErr)
+	}
+
+	err = tx.Commit()
+	return matchSentinelError(err)
+}
+
+// GetUsersInGroup searches for users that belong to a group and returns them.
+// Does not return ErrNotFound if none are found, as that is considered a
+// successful search.
+func (db *PgDB) GetUsersInGroup(ctx context.Context, gid int) ([]model.User, error) {
+	var users []model.User
+	err := Bun().NewSelect().Table("users").Model(&users).
+		Join("INNER JOIN user_group_membership AS ugm ON users.id=ugm.user_id").
+		Where("ugm.group_id = ?", gid).
+		Scan(ctx)
+
+	return users, err
+}
+
+func matchSentinelError(err error) error {
```

Comment:
nice. we could keep these helpers in `db` package for reuse

Reply:
Thanks! Yeah, I thought about that briefly. But I can sorta imagine it being possible different packages need to respond to codes differently. So for the time being decided to leave this specific to this package and then if we discover we're doing the same thing in several spots we can collect it into a centralized place. Totally willing to change my mind on this if y'all disagree that this seems prudent.